File: julia/binary-search/binary-search.jl
function binarysearch(arr, n)
    lo = 1
    hi = length(arr)
    while lo <= hi
        mid = Int(floor((lo + hi) / 2))
        if (arr[mid] < n)
            lo = mid + 1
        elseif (arr[mid] > n)
            hi = mid - 1
        elseif (arr[mid] == n)
            return mid:mid
        end
    end
    return lo:hi
end

File: lua/binary-search/binary-search.lua
return function(arr, search)
    local lo = 1
    local hi = #arr
    while lo <= hi do
        local mid = lo + math.floor((hi - lo) / 2)
        local tmp = arr[mid]
        if tmp == search then
            return mid
        elseif tmp < search then
            lo = mid + 1
        else
            hi = mid - 1
        end
    end
    return -1
end
